pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -93,6 +121,21 @@ func readDockercfg(filePath string) (cfg credentialprovider.DockerConfig, err er
 	return
 }
 
+// readDockerConfigJson reads the contents of a .docker/config.json file into a map
+// with server name keys and AuthEntry values
+func readDockerConfigJson(filePath string) (cfg credentialprovider.DockerConfig, err error) {
+	content, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return
+	}
+	var config credentialprovider.DockerConfigJson
```

Comment:
In this case, it's because it's passed as an interface{} -
https://github.com/golang/go/issues/10060

On Thu, Jan 28, 2016 at 10:00 PM, Clayton Coleman <ccoleman@redhat.com>
wrote:

> Well... it's not actually guaranteed to be on the heap if escape analysis
> shows its lifetime doesn't escape the block.  Because you passed & to
> fmt.Sprintf, it gets allocated on the heap because the compiler isn't
> positive that it doesn't get returned.  In other cases, it will stay on the
> stack and not the heap.
>
> On Thu, Jan 28, 2016 at 9:52 PM, Jordan Liggitt <notifications@github.com>
> wrote:
>
>> In pkg/build/builder/cmd/dockercfg/cfg.go
>> <https://github.com/openshift/origin/pull/6887#discussion_r51220567>:
>>
>> > @@ -93,6 +121,21 @@ func readDockercfg(filePath string) (cfg credentialprovider.DockerConfig, err er
>> >  	return
>> >  }
>> >
>> > +// readDockerConfigJson reads the contents of a .docker/config.json file into a map
>> > +// with server name keys and AuthEntry values
>> > +func readDockerConfigJson(filePath string) (cfg credentialprovider.DockerConfig, err error) {
>> > +	content, err := ioutil.ReadFile(filePath)
>> > +	if err != nil {
>> > +		return
>> > +	}
>> > +	var config credentialprovider.DockerConfigJson
>>
>> Yup. https://play.golang.org/p/02vGBl9PMh
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/openshift/origin/pull/6887/files#r51220567>.
>>
>
>